### PR TITLE
Polish paddle fluid framework error message - part2

### PR DESCRIPTION
**Pull request by chenwhql**
### PR types

Function optimization
### PR changes

Others
### Describe

Polish paddle fluid framework error message - part2

### CODE REVIEW

File: paddle/fluid/framework/ddim.cc [hunk 1]
```diff
@@ -81,9 +81,11 @@ bool contain_unknown_dim(const DDim& ddim) {
 }
 
 DDim slice_ddim(const DDim& dim, int begin, int end) {
-  PADDLE_ENFORCE(begin >= 0 && end <= dim.size(),
-                 "[begin(%d), end(%d)) must be inside [0, %d) in ddim slice.",
-                 begin, end, dim.size());
+  PADDLE_ENFORCE_EQ(
+      (begin >= 0 && end <= dim.size()), true,
+      platform::errors::InvalidArgument(
+          "[begin(%d), end(%d)) must be inside [0, %d) in ddim slice.", begin,
```
Comment (wanghuancoder) [inline paddle/fluid/framework/ddim.cc:87]: end(%d)) -> end(%d)

File: paddle/fluid/framework/dlpack_tensor.cc [hunk 2]
```diff
@@ -84,8 +89,8 @@ struct DLContextVisitor : public boost::static_visitor<::DLContext> {
     ctx.device_id = 0;
     return ctx;
 #else
-    PADDLE_THROW(
-        "platform::CUDAPinnedPlace is not supported in CPU only version");
+    PADDLE_THROW(platform::errors::Unavailable(
+        "platform::CUDAPinnedPlace is not supported in CPU only version"));
```
Comment (wanghuancoder) [inline paddle/fluid/framework/dlpack_tensor.cc]: 添加 .

File: paddle/fluid/framework/eigen.h [hunk 3]
```diff
@@ -26,7 +26,11 @@ struct EigenDim {
   using Type = Eigen::DSizes<Eigen::DenseIndex, D>;
 
   static Type From(const DDim& dims) {
-    PADDLE_ENFORCE(arity(dims) == D, "D must match arity(DDim)");
+    PADDLE_ENFORCE_EQ(arity(dims), D,
+                      platform::errors::InvalidArgument(
+                          "Input dimension size should be equal to %d, but "
+                          "received dimension size is %d.",
+                          D));
```
Comment (wanghuancoder) [inline paddle/fluid/framework/eigen.h]: D -> arity(dims), D

**Review by wanghuancoder — COMMENTED** [review]
有几个小问题，见具体回复。

File: paddle/fluid/framework/ddim.cc (hunk 1 above)
Comment (chenwhql) [reply paddle/fluid/framework/ddim.cc:87]: the last `)` is the pair of the first `[`

File: paddle/fluid/framework/dlpack_tensor.cc (hunk 2 above)
Comment (chenwhql) [reply paddle/fluid/framework/dlpack_tensor.cc]: done, thx

File: paddle/fluid/framework/eigen.h (hunk 3 above)
Comment (chenwhql) [reply paddle/fluid/framework/eigen.h]: done, thx

**Review by wanghuancoder — APPROVED** [review]
LGTM.

**Review by phlrain — APPROVED** [review]

**Pull request merged**